src/groupeddataframe/grouping.jl

```diff
@@ -122,50 +119,111 @@ _names(gd::GroupedDataFrame) = _names(gd.parent)
 # Wrapping automatically adds column names when the value returned
 # by the user-provided function lacks them
 wrap(x::Union{AbstractDataFrame, NamedTuple, DataFrameRow}) = x
-wrap(x::AbstractMatrix) = convert(DataFrame, x)
-wrap(x::AbstractVector) = DataFrame(x1 = x)
-wrap(x::Any) = (x1 = x,)
+wrap(x::AbstractMatrix) =
+    NamedTuple{Tuple(gennames(size(x, 2)))}(Tuple(x[:, i] for i in 1:size(x, 2)))
+wrap(x::Any) = (x1=x,)
+
+function do_call(f::Any, gd::GroupedDataFrame, incols::Tuple, i::Integer)
```

Comment:
here it seems to me that we are splatting columns in the call to `f`.

Reply:
Good catch. That's actually a legacy from the original design, but that function is only called with a one-element tuple, so it had no consequences. I've changed it to take a vector instead.

Reply:
OK